### PR TITLE
Use base64 url decoded for client secret when authenticating with Basic Auth

### PR DESCRIPTION
**Pull request by lexcao**
https://github.com/keycloak/keycloak/issues/11908

### CODE REVIEW

**Comment by frankjkelly** [comment]
Any update on this - can we get some 👁️ on this so we can get this merged eventually? Thanks!

**Review by mposolda — CHANGES_REQUESTED** [review]
@lexcao Thanks for the PR! Added one minor inline comment.

Will be also good to rebase this PR on top of latest keycloak main (maybe it will help to make sure that GH actions test will properly run with this PR as it seems they were not executed before for some reason).

File: core/src/main/java/org/keycloak/util/BasicAuthHelper.java [hunk 1]
```diff
@@ -18,32 +18,21 @@
...
-            throw new RuntimeException(e);
-        }
+public class BasicAuthHelper {
+    public static String createHeader(String username, String password) {
+        return "Basic " + Base64.encodeBytes((username + ':' + password).getBytes(StandardCharsets.UTF_8));
     }
 
-    public static String[] parseHeader(String header)
-    {
+    public static String[] parseHeader(String header) {
```
[the remark this answers is not in the document]
Comment (lexcao) [reply core/src/main/java/org/keycloak/util/BasicAuthHelper.java:35]: It could replace the entire method, I think.
Let me do that.

**Comment by lexcao** [comment]
Hi, @mposolda 
I have updated the PR.
Could you please help trigger the CI?

**Review by mposolda — APPROVED** [review]
@lexcao Thanks!

**Pull request merged**